From 136cc18a735081b12da58fbcfec5092fe8e7826d Mon Sep 17 00:00:00 2001
From: Rob Browning <rlb@defaultvalue.org>
Date: Fri, 8 Apr 2011 22:50:51 -0500
Subject: Fix a tty-related preprocessor inclusion error affecting GNU/Hurd
 systems.

A problem with the preprocessor defines in src/term.c has been fixed.
This problem would cause Emacs to fail on GNU/Hurd systems as follows:

  $ emacs -nw
  emacs: Not a tty device: /dev/tty

Author: Samuel Thibault <sthibault@debian.org>
Date: Thu, 20 Jan 2011 02:17:24 +0100
Applied-By: Rob Browning <rlb@defaultvalue.org>
Forwarded: bug-gnu-emacs@gnu.org
Closes: #610576
---
 src/term.c |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/src/term.c b/src/term.c
index 6d16a65..44e127e 100644
--- a/src/term.c
+++ b/src/term.c
@@ -3413,13 +3413,12 @@ init_tty (char *name, char *terminal_type, int must_succeed)
          if we don't have one at the moment.  */
       fd = emacs_open (name, O_RDWR | O_IGNORE_CTTY | O_NOCTTY, 0);
     else
-#else
+#endif /* O_IGNORE_CTTY */
       /* Alas, O_IGNORE_CTTY is a GNU extension that seems to be only
          defined on Hurd.  On other systems, we need to explicitly
          dissociate ourselves from the controlling tty when we want to
          open a frame on the same terminal.  */
       fd = emacs_open (name, O_RDWR | O_NOCTTY, 0);
-#endif /* O_IGNORE_CTTY */
 
     tty->name = xstrdup (name);
     terminal->name = xstrdup (name);
-- 
1.7.6.3

